core/src/main/java/hudson/model/Node.java

```diff
@@ -451,6 +451,47 @@ public FileSystemProvisioner getFileSystemProvisioner() {
      */
     public abstract @Nonnull DescribableList<NodeProperty<?>, NodePropertyDescriptor> getNodeProperties();
 
+    /**
+     * Gets the specified property or null if the property is not configured for this Node.
+     * 
+     * @param clazz the type of the property
+     * 
+     * @return null if the property is not configured
+     * 
+     * @since TODO
+     */
+    @CheckForNull
+    public <T extends NodeProperty> T getNodeProperty(Class<T> clazz)
```

Comment:
Uh, sorry. For the parameter it may be reasonable, but Nonnull is implied in such case in Oracle code guidelines being used by Jenkins. So it is not really must-have here even though FindBugs does not process the code according to Oracle code style now